src/main/cpp/src/get_json_object.cu

```diff
@@ -70,14 +68,49 @@ struct path_instruction {
   cudf::string_view name;
 
   // used when type is index
-  int index{-1};
+  int64_t index{-1};
 };
 
-// TODO parse JSON path
-thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
-  cudf::string_scalar const& json_path)
+rmm::device_uvector<path_instruction> construct_path_commands(
+  std::vector<std::tuple<path_instruction_type, std::string, int64_t>> const& instructions,
+  cudf::string_scalar const& all_names_scalar,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
 {
-  return thrust::nullopt;
+  int name_pos = 0;
+
+  // construct the path commands
+  std::vector<path_instruction> path_commands;
+  for (auto const& inst : instructions) {
+    auto const& [type, name, index] = inst;
+    switch (type) {
+      case path_instruction_type::subscript:
+        path_commands.emplace_back(path_instruction{path_instruction_type::subscript});
+        break;
+      case path_instruction_type::wildcard:
+        path_commands.emplace_back(path_instruction{path_instruction_type::wildcard});
+        break;
+      case path_instruction_type::key:
+        path_commands.emplace_back(path_instruction{path_instruction_type::key});
+        path_commands.back().name =
+          cudf::string_view(all_names_scalar.data() + name_pos, name.size());
+        name_pos += name.size();
+        break;
+      case path_instruction_type::index:
+        path_commands.emplace_back(path_instruction{path_instruction_type::index});
+        path_commands.back().index = index;
+        break;
+      case path_instruction_type::named:
+        path_commands.emplace_back(path_instruction{path_instruction_type::named});
+        path_commands.back().name =
+          cudf::string_view(all_names_scalar.data() + name_pos, name.size());
+        name_pos += name.size();
+        break;
+      default: CUDF_FAIL("Invalid path instruction type");
+    }
+  }
+  // convert to uvector
+  return cudf::detail::make_device_uvector_sync(path_commands, stream, mr);
```

Comment:
I doubt if this can work for std::string type.
This method is copying a memory block, but string has a pointer.
I suspect it only copy a pointer and does not copy the data that pointer refers to.
Need to test in kernel code.